test/e2e-go/restAPI/restClient_test.go

```diff
@@ -1051,3 +1051,160 @@ return
 	a.NoError(err)
 	a.Equal(expectedMetadata, *createdAssetInfo.Params.MetadataHash)
 }
+
+func TestStateProofInParticipationInfo(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	a := require.New(fixtures.SynchronizedTest(t))
+	var localFixture fixtures.RestClientFixture
+
+	proto := config.Consensus[protocol.ConsensusCurrentVersion]
+	proto.EnableStateProofKeyregCheck = true
+	localFixture.SetConsensus(config.ConsensusProtocols{protocol.ConsensusCurrentVersion: proto})
+
+	localFixture.Setup(t, filepath.Join("nettemplates", "TwoNodes50Each.json"))
+	defer localFixture.Shutdown()
+
+	testClient := localFixture.LibGoalClient
+	waitForRoundOne(t, testClient)
+	wh, err := testClient.GetUnencryptedWalletHandle()
+	a.NoError(err)
+	addresses, err := testClient.ListAddresses(wh)
+	a.NoError(err)
+	_, someAddress := getMaxBalAddr(t, testClient, addresses)
+	a.NotEmpty(someAddress, "no addr with funds")
+
+	addr, err := basics.UnmarshalChecksumAddress(someAddress)
+	a.NoError(err)
+
+	params, err := testClient.SuggestedParams()
+	a.NoError(err)
+
+	firstRound := basics.Round(params.LastRound + 1)
+	lastRound := basics.Round(params.LastRound + 1000)
+	dilution := uint64(100)
+	randomVotePKStr := randomString(32)
+	var votePK crypto.OneTimeSignatureVerifier
+	copy(votePK[:], []byte(randomVotePKStr))
+	randomSelPKStr := randomString(32)
+	var selPK crypto.VRFVerifier
+	copy(selPK[:], []byte(randomSelPKStr))
+	var gh crypto.Digest
+	copy(gh[:], params.GenesisHash)
+
+	tx := transactions.Transaction{
+		Type: protocol.KeyRegistrationTx,
+		Header: transactions.Header{
+			Sender:      addr,
+			Fee:         basics.MicroAlgos{Raw: 10000},
+			FirstValid:  firstRound,
+			LastValid:   lastRound,
+			GenesisHash: gh,
+		},
+		KeyregTxnFields: transactions.KeyregTxnFields{
+			VotePK:      votePK,
+			SelectionPK: selPK,
+			VoteFirst:   firstRound,
+			StateProofPK: merklekeystore.Verifier{
+				Root: [merklekeystore.KeyStoreRootSize]byte{1, 2, 3, 4},
```

Comment:
It would be slightly better to use a randomly generated 64 byte string for this, like the other participation keys, so we can be sure all bytes beyond the first 4 are being handled properly.